docs/painless/painless-contexts/painless-bucket-script-agg-context.asciidoc

```diff
@@ -0,0 +1,87 @@
+[[painless-bucket-script-agg-context]]
+=== Bucket Script aggregation context
+
+Use a Painless script in an
+{ref}/search-aggregations-pipeline-bucket-script-aggregation.html[`bucket_script` pipeline aggregation]
+to calculate a value as a result in a bucket.
+
+==== Variables
+
+`params` (`Map`, read-only)::
+        User-defined parameters passed in as part of the query. The parameters
+        include values defined as part of the `buckets_path`.
+
+==== Return
+
+numeric::
+        The calculated value as the result.
+
+==== API
+
+The standard <<painless-api-reference, Painless API>> is available.
+
+==== Example
+
+To run this example, first follow the steps in <<painless-context-examples, context examples>>.
+
+The painless context in a `bucket_script` aggregation provides a `params` map.  This map contains both
+user-specified custom values, as well as the values from other aggregations specified in the `buckets_path`
+property.
+
+For our example, we are going to take the values from a `min` and `max` aggregation, subtract them to find
+the difference, then add in a "base cost" from a user-specified value.  The script looks like this:
+
+[source,Painless]
+--------------------------------------------------
+(params.max - params.min) + params.base_cost
+--------------------------------------------------
+
+Note how all the values are being pulled from the `params` map.  In context, the aggregation looks like this:
+
+[source,js]
+--------------------------------------------------
+GET /seats/_search
+{
+  "size": 0,
+  "aggs": {
+    "theatres": {
+      "terms": {
+        "field": "theatre",
+        "size": 10
+      },
+      "aggs": {
+        "min_cost": {
+          "min": {
+            "field": "cost"
+          }
+        },
+        "max_cost": {
+          "max": {
+            "field": "cost"
+          }
+        },
+        "spread_plus_base": {
+          "bucket_script": {
+            "buckets_path": { <1>
+              "min": "min_cost",
+              "max": "max_cost"
+            },
+            "script": {
+              "params": {
+                "base_cost": 5 <2>
+              },
+              "source": "(params.max - params.min) + params.base_cost"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:seats]
+<1> the `buckets_path` points to two aggregations (`min_cost`, `max_cost`) and adds `min`/`max` variables
+to the `params` map
+<2> the user-specified `base_cost` is added in the script's `params` section, which is also added to the
+`params` variable for use in the script.
```

Comment:
I think we generally (but likely not consistently) init-cap callouts whether or not they are complete sentences.

In <2>, I'd refer to it as the `params` map for consistency. I think I'd also drop "for use in the script".